pkg/backend/httpstate/client/api.go

```diff
@@ -45,6 +45,9 @@ const (
 	apiRequestDetailLogLevel = 11 // log level for logging extra details about API requests and responses
 )
 
+// ErrLoginRequired is returned when a command requires logging in.
+var ErrLoginRequired = errors.New("this command requires logging in; try running `pulumi login` first")
```

Comment:
We have essentially the same error, once in the `client` package, and once `backend`. We don't (and can't) have `backend` depend on `client` due to cycles. Maybe this error type should live elsewhere? I maintained the same wording, since this will sometimes be shown to the user.